### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by yezyilomo**
- pass the whole context instead of extracting request and add it to the context
- Makes `has_query_param`, `get_raw_query`, and `get_parsed_query_from_req` class methods so that they can be reused on a view to implement dynamically prefetching and selecting related fields.

### CODE REVIEW

**Comment by yezyilomo** [comment]
Hey @tredzko what do you think of this?, I've refactored `DynamicFieldsMixin` so that we can reuse `has_query_param`, `get_raw_query`, and `get_parsed_query_from_req` on `RestQLViewMixin`, I also think they were supposed to be class methods from the beginning because they were not using instance.

File: tests/testapp/views.py [hunk 1]
```diff
@@ -20,6 +20,8 @@ class BookViewSet(viewsets.ModelViewSet):
 
 
 class CourseViewSet(viewsets.ModelViewSet):
+	select_related = []
+	prefetch_related = ["books"]
```
[the remark this answers is not in the document]
Comment (yezyilomo) [reply tests/testapp/views.py]: Oooh yeah I forgot to remove those.

**Comment by tredzko** [comment]
@yezyilomo That makes a lot of sense. If this gets in I'll refactor how we check for the query param and get parsed values on that view mixin. Do you prefer I rebase or merge?

File: django_restql/fields.py [hunk 2]
```diff
@@ -60,14 +60,12 @@ def validate_pk_list(self, pks):
...
-            request = self.context.get('request')
-            context = {"request": request}
             model = self.parent.Meta.model
-            rel = getattr(model, self.source).rel
+            rel = getattr(model, self.field_name).rel
 
             if isinstance(rel, ManyToOneRel):
                 # ManyToOne Relation
-                field_name = getattr(model, self.source).field.name
+                field_name = getattr(model, self.field_name).field.name
```
Comment (tredzko) [inline django_restql/fields.py]: Not sure if this is needed since we have `self.field_name`. Can `field.name` be different?

Comment (yezyilomo) [reply django_restql/fields.py]: You mean if `getattr(model, self.source).field.name` will be different from `getattr(model, self.source)`?

> @yezyilomo That makes a lot of sense. If this gets in I'll refactor how we check for the query param and get parsed values on that view mixin. Do you prefer I rebase or merge?

Just merge.

Comment (tredzko) [reply django_restql/fields.py]: Gotcha. And I agree that source is different, but I mean is `getattr(model, self.field_name).field.name` different from `self.field_name`?

Comment (yezyilomo) [reply django_restql/fields.py]: Yeah it's different.

**Pull request merged**